synapse/telepath.py

```diff
@@ -103,15 +105,30 @@ def zipurl(info):
 
     return url
 
+def modurl(url, **info):
+    if isinstance(url, str):
+        urlinfo = chopurl(url)
+        for k, v in info.items():
+            urlinfo[k] = v
+        return zipurl(urlinfo)
+    return [modurl(u, **info) for u in url]
+
 def mergeAhaInfo(info0, info1):
 
+    # info0 - local urlinfo
+    # info1 - urlinfo provided by aha
+
     # copy both to prevent mutation
     info0 = dict(info0)
```

Comment:
These should probably be copy.deepcopy() calls to fully isolate them, in the event of mutable values.

Reply:
There currently are no mutable values within telepath urlinfo dictionaries.  They are just flat key/values.